### PR TITLE
Adding UUID to App Configuration requests

### PR DESCRIPTION
**Pull request by mrm9084**
## Description
Added a new header onto App Configurations requests setting the x-ms-client-request-id.

## Related PRs
List related PRs against other branches: PR 4752955

## Steps to Test
Have a logback.xml file in your applications resource folder with the level set to DEBUG. When requests are made the UUID will be outputted to the applications logs, which can then be matched to an App Configuration log.

### CODE REVIEW

**Comment by zhenlan** [comment]
cc: @sophiaso @MSGaryWang @jimmyca15

File: spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigHttpClient.java [hunk 1]
```diff
@@ -69,6 +69,17 @@ public CloseableHttpResponse execute(@NonNull HttpUriRequest request, Date date,
         return httpClient.execute(request);
     }
 
+    /**
+     * Generates a request Header with a date, client request id, and a sha256 content hash.
```
Comment (zhenlan) [inline spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigHttpClient.java]: NIT: technically this generates multiple headers... not "*a* request Header..."

File: spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigHttpClient.java [hunk 2]
```diff
@@ -69,6 +69,17 @@ public CloseableHttpResponse execute(@NonNull HttpUriRequest request, Date date,
...
+     * 
+     * @param request the request that will be sent with this header and will have a hash generated for it.
+     * @param date the current date and time
+     * @param credential HMAC-SHA256 Credential
```
Comment (zhenlan) [inline spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigHttpClient.java]: credential is "Access key ID"

File: spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigHttpClient.java [hunk 3]
```diff
@@ -69,6 +69,17 @@ public CloseableHttpResponse execute(@NonNull HttpUriRequest request, Date date,
...
+     * @param secret Key to encode HMAC
```
Comment (zhenlan) [inline spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigHttpClient.java]: secret is "Access key value"

File: spring-cloud-azure-appconfiguration-config/src/test/java/com/microsoft/azure/spring/cloud/config/ConfigHttpClientTest.java [hunk 4]
```diff
@@ -105,22 +133,27 @@ public void setup() {
...
     public void testRequestHeadersForAuthConfigured() throws URISyntaxException, IOException {
         assertThat(REQUEST_DATA_LIST.size()).isEqualTo(4);
+        PowerMockito.mockStatic(UUID.class);
+        PowerMockito.when(UUID.randomUUID()).thenReturn(uuid1, uuid2, uuid3, uuid4);
         for (int index = 0; index < REQUEST_DATA_LIST.size(); index++) {
             TestRequestData reqData = REQUEST_DATA_LIST.get(index);
 
             CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+            
+
```
Comment (zhenlan) [inline spring-cloud-azure-appconfiguration-config/src/test/java/com/microsoft/azure/spring/cloud/config/ConfigHttpClientTest.java]: NIT: extra empty line

File: spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigHttpClient.java [hunk 5]
```diff
@@ -39,7 +39,8 @@
  * How to use:
  * <p>
  * HttpGet httpGet = new HttpGet("https://my-config-store.azconfig.io/keys");
- * CloseableHttpResponse response = ConfigHttpClient.execute(httpGet, "my-credential", "my-secret");
+ * CloseableHttpResponse response = ConfigHttpClient.execute(httpGet, "my-credential",
```
Comment (MSGaryWang) [inline spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigHttpClient.java]: nit: just want to know why we change line here?

Comment (mrm9084) [reply spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigHttpClient.java]: I think it was just an auto format that did that. I will put it back.

**Pull request merged**